### PR TITLE
HotFix: SkyRegions GUI

### PR DESCRIPTION
**Pull request by rcooke-ast**
This hotfix ensures that the calib key/dir/ID are all stored in the SkyRegions calibrations file. Several errors and bits of code cleanup are also implemented.

### CODE REVIEW

File: pypeit/core/gui/skysub_regions.py [hunk 1]
```diff
@@ -571,22 +571,41 @@ def operations(self, key, axisID):
...
-                msgs.info(f"Saving regions to: {outfil}")
-                self._overwrite = True
-            msskyreg = buildimage.SkyRegions(image=inmask.astype(float), PYP_SPEC=self.spectrograph)
-            msskyreg.to_file(file_path=outfil)
+            if np.all(np.logical_not(inmask)):
+                msgs.warn("Sky regions are empty - master calibration frame will not be generated")
+            else:
+                # Build the master Sky Regions calibration frame
+                msskyreg = buildimage.SkyRegions(image=inmask.astype(float), PYP_SPEC=self.spectrograph)
+        return msskyreg
```
Comment (kbwestfall) [inline pypeit/core/gui/skysub_regions.py]: I think this is a bit clearer:
```
if not self._use_updates:
    return None

# Generate the mask
inmask = skysub.generate_mask(self.pypeline, self._skyreg, self.slits, self.slits_left,
                              self.slits_right)
if np.all(np.logical_not(inmask)):
    msgs.warn("Sky regions are empty - master calibration frame will not be generated")
    return None

# Build the master Sky Regions calibration frame
return buildimage.SkyRegions(image=inmask.astype(float), PYP_SPEC=self.spectrograph)

```

**Review by kbwestfall — APPROVED** [review]
One minor suggestion, but it looks good otherwise.  This touches a core function, so should we run tests?

File: pypeit/core/gui/skysub_regions.py (hunk 1 above)
Comment (rcooke-ast) [reply pypeit/core/gui/skysub_regions.py]: Agreed - I've made that change!

>

**Comment by rcooke-ast** [comment]
> This touches a core function, so should we run tests?

Yes, we probably should - running now...

**Comment by rcooke-ast** [comment]
Two dev suite fails, but these are unrelated to this PR. One is fixed by #1710, and the other is fixed by #1712.

image

File: pypeit/core/gui/skysub_regions.py [hunk 2]
```diff
@@ -571,22 +571,41 @@ def operations(self, key, axisID):
         self.replot()
 
     def get_result(self):
-        """Save a mask containing the skysub regions, and print information
+        """Generate a master calibration file containing a mask of the skysub regions, and print information
```
Comment (profxj) [inline pypeit/core/gui/skysub_regions.py]: please remove `master`

File: pypeit/core/gui/skysub_regions.py [hunk 3]
```diff
@@ -571,22 +571,41 @@ def operations(self, key, axisID):
...
+        if not self._use_updates:
+            return None
+
+        # Generate the mask
+        inmask = skysub.generate_mask(self.pypeline, self._skyreg, self.slits, self.slits_left, self.slits_right)
+        if np.all(np.logical_not(inmask)):
+            msgs.warn("Sky regions are empty - master calibration frame will not be generated")
+            return None
+
+        # Build the master Sky Regions calibration frame
```
Comment (profxj) [inline pypeit/core/gui/skysub_regions.py]: same

**Review by profxj — APPROVED** [review]
all good, modulo the `master` bit
  time to scrub that from our past  :)

File: pypeit/core/gui/skysub_regions.py (hunk 2 above)
Comment (rcooke-ast) [reply pypeit/core/gui/skysub_regions.py]: Done (and I've also removed some other instances of this, as well)!

File: pypeit/core/gui/skysub_regions.py (hunk 3 above)
Comment (rcooke-ast) [reply pypeit/core/gui/skysub_regions.py]: Done!

**Pull request merged**